Chapters/01-zig-weird.qmd

```diff
@@ -243,25 +259,27 @@ care about this keyword.
 Now that we have learned a lot about Zig's syntax from the `root.zig` file,
 let's take a look at the `main.zig` file.
 A lot of the elements we saw in `root.zig` are also present in `main.zig`.
-But we have some other elements that we did not have seen yet, so let's dive in.
+But there are some other elements that we haven't seen yet, so let's dive in.
 
 First, look at the return type of the `main()` function in this file.
-We can see a small change. Now, the return
+We can see a small change. The return
 type of the function (`void`) is accompanied by an exclamation mark (`!`).
-What this exclamation mark is telling us, is that this `main()` function
-might also return an error.
-
-So, in this example, the `main()` function can either return `void`, or, return an error.
-This is an interesting feature of Zig. If you write a function, and, something inside of
-the body of this function might return an error, then, you are forced to:
-
-- either add the exclamation mark to the return type of the function, to make it clear that
-this function might return an error.
-- or explicitly handle this error that might occur inside the function, to make sure that,
-if this error does happen, you are prepared, and your function will no longer return an error
-because you handled the error inside your function.
-
-In most programming languages, we normally handle (or deals with) an error through
+This exclamation mark tells us that this `main()` function
+might return an error.
+
+In this example, the `main()` function can either return `void` or return an error.
+This is an interesting feature of Zig. If you write a function and something inside of
+the body of this function might return an error then you are forced to:
+
+- either add the exclamation mark to the return type of the function and make it clear that
+this function might return an error
+- explicitly handle this error inside the function
+
+<!-- is it different in Zig or not?
+C++ has try/catch
+Zig has try if i understand correctly. is there a catch somewhere?

```

Comment:
Uhmm.... Zig has both try and catch. But I see what you mean. Saying that Zig has *try catch pattern* passes the wrong message.